### ISSUE TITLE
WIP: Top-level polymorphic schema definition

### ISSUE THREAD

**Issue by jdegoes** [scalaz/scalaz-schema#7]
Since we are using `SchemaModule`, we may want to create a top-level, polymorphic definition of a schema. Something like this:

```scala
trait Schema[A] {
  def apply(m: SchemaModule): m.Schema[A]
}
```

Then a user can write:

```scala
object PersonSchema extends Schema[Person] {
  def apply(m: SchemaModule): m.Schema[A] = {
    import m._

    // Use everything in `m`
  }
}
```

If now we had a way to abstract over the primitives required by a schema, it would then be possible to have polymorphic schemas (across, e.g., Scala, JSON, XML, etc.).

**Comment by vil1** [comment]
This is (indirectly) addressed by #27

**Comment by vil1** [comment]
This was solved/made irrelevant by #27
